webpack.config.ts

```diff
@@ -1,33 +1,46 @@
 /* eslint-env node */
 /* eslint import/no-nodejs-modules:0 */
-const fs = require('fs');
-const path = require('path');
-
-const {CleanWebpackPlugin} = require('clean-webpack-plugin'); // installed via npm
-const webpack = require('webpack');
-const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
-const MiniCssExtractPlugin = require('mini-css-extract-plugin');
-const CompressionPlugin = require('compression-webpack-plugin');
-const FixStyleOnlyEntriesPlugin = require('webpack-remove-empty-scripts');
-const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
-
-const IntegrationDocsFetchPlugin = require('./build-utils/integration-docs-fetch-plugin');
-const SentryInstrumentation = require('./build-utils/sentry-instrumentation');
-const LastBuiltPlugin = require('./build-utils/last-built-plugin');
-const babelConfig = require('./babel.config');
+
+import fs from 'fs';
+import path from 'path';
+
+import {CleanWebpackPlugin} from 'clean-webpack-plugin';
+import CompressionPlugin from 'compression-webpack-plugin';
+import CssMinimizerPlugin from 'css-minimizer-webpack-plugin';
+import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+import webpack from 'webpack';
+import {Configuration as DevServerConfig} from 'webpack-dev-server';
+import FixStyleOnlyEntriesPlugin from 'webpack-remove-empty-scripts';
+
+import IntegrationDocsFetchPlugin from './build-utils/integration-docs-fetch-plugin';
+import LastBuiltPlugin from './build-utils/last-built-plugin';
+import SentryInstrumentation from './build-utils/sentry-instrumentation';
+import babelConfig from './babel.config';
+
+/**
+ * Merges the devServer config into the webpack config
+ *
+ * See: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/43232
+ */
+interface Configuration extends webpack.Configuration {
```

Comment:
we don't usually use interface, should we use type here?

```
type Configuration = webpack.Configuration & {
  devServer?: DevServerConfig;
};
```

Reply:
I'm going to leave it as an interface since I copied this out of that that particular issue. At some point this will go away.